tools/docker/Run-ContainerTests.ps1

```diff
@@ -0,0 +1,278 @@
+      $containerIsRunning = $null -ne (docker ps --filter name=$Container -q)
+
+      if ($containerIsRunning) {
+         docker stop $containerId
+      }
+
+      if ($containerId) {
+         docker rm $Container
+      }
+
+      $psCommandString = ($pesterBuild.ToString()) -replace '#Container#', $Container
+
+      docker run `
+         -dit `
+         --name  $Container `
+         --volume $Volume `
+         -w $DefaultWorkDir `
+         $Image `
+         $Shell -Command $psCommandString
+
+      if ($FollowLogs) {
+         $output = (docker exec -it $Container $Shell -c '$PSVersionTable | ConvertTo-Json -Compress;') -join ''
+         $outputFirst = $output.IndexOf('{')
+         $ouputLast = $output.LastIndexOf('}')
+         $versiontable = $output.Substring($outputFirst, $ouputLast+1-$outputFirst) | ConvertFrom-Json -Depth 50
+         $psVersion = "$($versiontable.PSVersion.Major).$($versiontable.PSVersion.Minor).$($versiontable.PSVersion.Patch)"
+
+         $argList = "-NoExit -Command `"`$Host.UI.RawUI.WindowTitle = 'VSTeam Unit Tests | PowerShell $($versiontable.PSEdition) $psVersion | $($versiontable.Os)'; docker logs $Container -f`""
+         Start-Process $Shell -argumentlist $argList
+      }
+
+      if ($Wait) {
+         docker wait $Container
+      }
+
+   }
+}
+
+function Wait-DockerContainer {
+   <#
+    .SYNOPSIS
+        Wait for the given containers to finish. If they don't run, then error is thrown
+   #>
+   [CmdletBinding()]
+   param (
+      # Containers to wait for
+      [Parameter(Mandatory = $true)]
+      [string[]]
+      $Container
+   )
+
+   process {
+
+      $exitCodes = @()
+
+      $runningContainers = docker ps --format '{{json .}}' | ConvertFrom-Json
+
+      $notRunningContainers = @()
+      $notAllContainersRunning = $Container.Count -ne ($Container | Where-Object {
+
+            $contains = $runningContainers.Names.Contains($_)
+            if ($contains) {
+               return $true
+            }
+            else {
+               $notRunningContainers += $_
+               return $false
+            }
+
+         }).Count
+
+      if ($notAllContainersRunning) {
+         Write-Error "Contains with the following names are not running: $($notRunningContainers -join ', ')"
+      }
+      else {
+
+         $Container | ForEach-Object {
+            $exitCode = docker wait $_
+
+            $exitCodes += @{
+               containerName = $_
+               exitCode = $exitCode
+            }
+         }
+      }
+
+      return $exitCodes
+   }
+}
+
+$platform = $PSVersionTable.Platform
+if ($platform -ne "Win32NT" -or $UseLinux) {
+   Write-Warning "Platform is not Win32NT based but '$platform'. Windows container do not work on linux based systems. Ignoring windows containers..."
+}
+
+$scriptPath = $PSScriptRoot
+$rootDir = (Resolve-Path -Path "$scriptPath\..\..\").ToString().trim('\')
+$containerFolder = "c:/vsteam"
+$containerFilePath = "$rootDir/build/docker"
```

Comment:
this should be "$rootDir/Tools/docker" not build/docker.

Reply:
Also fixed the dir. I moved it before.